include/_atarios.h

```diff
@@ -569,39 +587,40 @@ struct __os {
                     unsigned char daux2;    // = $030B          1-byte second command auxiliary
                 };
             };
-            unsigned int timer1;            // = $030C/$030D    INITIAL TIMER VALUE
-#ifdef OSA           
-            unsigned char addcor;           // = $030E          ##old## ADDITION CORRECTION
-#else           
-            unsigned char jmpers;           // = $030E          ##1200xl## 1-byte jumper options
-#endif
-            unsigned char casflg;           // = $030F          CASSETTE MODE WHEN SET
-            unsigned char timer2;           // = $0310/$0311    2-byte final baud rate timer value
-            unsigned char temp1;            // = $0312          TEMPORARY STORAGE REGISTER
-#ifdef OSA       
-            unsigned char _spare_5;         // = $0313          unused
-            unsigned char temp2;            // = $0314          ##old## TEMPORARY STORAGE REGISTER
-#else                   
-            unsigned char temp2;            // = $0313          ##1200xl## 1-byte temporary
-            unsigned char ptimot;           // = $0314          ##1200xl## 1-byte printer timeout
-#endif      
-            unsigned char temp3;            // = $0315          TEMPORARY STORAGE REGISTER
-            unsigned char savio;            // = $0316          SAVE SERIAL IN DATA PORT
-            unsigned char timflg;           // = $0317          TIME OUT FLAG FOR BAUD RATE CORRECTION
-            unsigned char stackp;           // = $0318          SIO STACK POINTER SAVE CELL
-            unsigned char tstat;            // = $0319          TEMPORARY STATUS HOLDER
-#ifdef OSA   
-            HATABS hatabs[12];              // = $031A-$033D    handler address table
-            unsigned int zeropad;           // = $033E/$033F    zero padding
-#else
-            HATABS hatabs[11];              // = $031A-$033A    handler address table
-            unsigned int zeropad;           // = $033B/$033C    zero padding
-            unsigned char pupbt1;           // = $033D          ##1200xl## 1-byte power-up validation byte 1
-            unsigned char pupbt2;           // = $033E          ##1200xl## 1-byte power-up validation byte 2
-            unsigned char pupbt3;           // = $033F          ##1200xl## 1-byte power-up validation byte 3
-#endif
         };
     };
+    unsigned int timer1;                    // = $030C/$030D    INITIAL TIMER VALUE
+#ifdef OSA                      
+    unsigned char addcor;                   // = $030E          ##old## ADDITION CORRECTION
+#else                       
+    unsigned char jmpers;                   // = $030E          ##1200xl## 1-byte jumper options
+#endif          
```

Comment:
Regarding the `#ifdef OSA`: I understand that you added it for completeness. And that's fine.

But how would one use these definitions? I guess nowadays no one in his right mind would write a program which only works on the OSA ROM version. So the program would have to to adapt at runtime to the OS version.

Wouldn't it be better to make unions of the OSA and later version definitions? In order that both the OSA and later definitions are readily at hand?

Reply:
> I understand that you added it for completeness.
Yes, that was exactly my motivation.

>Wouldn't it be better to make unions of the OSA and later version definitions?
I don't think so. Currently you actively have to decide to service OSA. Offering a union would allow  usage by accident (e.g. wrong documentation or false understanding of the OS model.)